src/snowflake/snowpark/_internal/analyzer/snowflake_plan_node.py

```diff
@@ -100,12 +100,12 @@ def __init__(
         # When `is_temp_table_for_cleanup` is True, it's a temp table
         # generated by Snowpark (currently only df.cache_result) under the hood
         # and users are not aware of it.
-        if is_temp_table_for_cleanup:
-            session._temp_table_ref_count_map[name] += 1
+        if is_temp_table_for_cleanup and session._temp_table_cleaner is not None:
+            session._temp_table_cleaner.inc_ref_count(name)
```

Comment:
can you move all  those to the temp_table_cleaner, you can have an interface 
```
table_ref_created(table: SnowflakeTable) {
    inc_ref_count(name)
    table._finalizer = weakref.finalize(dec_ref_count, name)
}
```

Reply:
good call, I moved all logic to temp_table_cleaner now, which is much neater